public/scripts/todoService.js

```diff
@@ -0,0 +1,18 @@
+angular
+  .module('MyTodoApp')
+  .service('TodoService', function() {
+
+    this.todoList = ['Demo Task'];
+
+    this.completedList = [];
+
+    this.addTask = function(newTask) {
+      this.todoList.push(newTask);
+    };
+
+    this.completeTask = function(item, index) {
```

Comment:
This is kinda an iterator pattern - to access both `item` and `index` in an enumerator, but not really necessary here as `completeTask()` works on a single item